### DISCUSSION TITLE
Zig Source Code Highlighting

### DISCUSSION THREAD

**Discussion by mkashirin** [community/community#111295] [Other Feature Feedback, Questions, & Ideas] [answered]
### Question

Hi!

I recently started to learn the **Zig** programming language. But when browsing the source code of the language, the highlighting confuses me alot. There's too much **orange** in it (structs, enums, functions and even variables are orange!).

I am wondering if there is some particular reason for this? If so, what is it? If there isn't any, why not just make functions' and variables' names white?

**Comment by lin20260407** [comment]
"Hi!

The use of orange for various elements like structs, enums, functions, and variables in the Zig programming language's source code highlighting isn't a characteristic of Zig itself but is determined by the syntax highlighting theme of your text editor or IDE. Syntax highlighting is designed to enhance readability by distinguishing different code elements with specific colors. The choice of orange in your case is likely a default setting or a specific theme you've chosen.

The reason for not making functions' and variables' names white by default could be due to the theme designer's preference for color schemes that differentiate code elements more distinctly. However, if you find the abundance of orange confusing or prefer a different color scheme, most editors and IDEs allow you to customize the theme. You can adjust the colors used for syntax highlighting to your liking, potentially changing functions and variables to white or any other color you prefer.

This flexibility in customization is meant to cater to diverse preferences and enhance the coding experience by making the code visually easier to navigate and understand."

**Comment by spenserblack** [comment, marked as answer]
[Linguist](https://github.com/github-linguist/linguist) is the tool that GitHub uses to not only identify languages, but provide the appropriate grammar for syntax highlighting for that language. Looking at Linguist, you can eventually find that the grammar that is used is [ziglang/sublime-zig-language](https://github.com/ziglang/sublime-zig-language/tree/1a4a38445fec495817625bafbeb01e79c44abcba).If you look into that repo, you might be able to find why so much of the syntax is highlighted the same color.

If they decide to update the grammar to improve highlighting, you should be able to then make a PR for linguist, which will eventually result in the syntax highlighting being updated on GitHub.
